### PR TITLE
fixes #5722: VM-based host cloning is functional again now.

### PR DESCRIPTION
**Pull request by dmitri-d**

### CODE REVIEW

**Comment by dmitri-d** [comment]
[test]

**Comment by dmitri-d** [comment]
[test]

**Comment by dmitri-d** [comment]
[test]

File: app/models/host/managed.rb [hunk 1]
```diff
@@ -628,6 +628,9 @@ def clone
     # do not copy system specific attributes
     host = self.dup(:include => [:host_config_groups, :host_classes, :host_parameters],
                     :except  => [:name, :mac, :ip, :uuid, :certname, :last_report])
+    if self.compute_resource
```
Comment (isratrade) [inline app/models/host/managed.rb:631]: I would also add here `compute_profile_id`

``` ruby
if compute_resource && compute_profile_id
```

**Comment by isratrade** [comment]
@witlessbird, when I test this in the UI using the develop branch, the "Virtual Machine" tab includes the attributes correctly per the compute profile.  Can you reproduce the case if which the "Virtual Machine" doesn't populate correctly.

**Comment by dmitri-d** [comment]
Nods; I can reproduce it.

File: app/models/host/managed.rb (hunk 1 above)
Comment (abenari) [inline app/models/host/managed.rb:631]: compute profile id is not needed here, as we need to get the actual vm details not the profile. the profile can be overridden.

File: app/models/host/managed.rb [hunk 2]
```diff
@@ -628,6 +628,9 @@ def clone
...
+      host.compute_attributes = host.compute_resource.compute_profile_attributes_for(self.compute_profile_id)
```
Comment (abenari) [inline app/models/host/managed.rb]: I think this implementation has a problem, sometimes compute profiles are not in use, sometimes after selecting the compute profile the vm attributes are being edited. I think we must use the vm attributes when cloning
I suggest the following code:

``` ruby
host.compute_attributes = compute_resource.compute_attributes_for(uuid)
```

and implement compute_attributes_for in the app/model/compute_resource.rb

``` ruby
def compute_attributes_for(uuid)
  vm = find_vm_by_uuid(uuid)
  vm.attributes.select{|k,v| k != :id }
end
```

Comment (dmitri-d) [inline app/models/host/managed.rb]: good catch, @abenari

**Comment by dmitri-d** [comment]
@isratrade, @abenari: ping?

**Comment by isratrade** [comment]
@witlessbird, looks good. I agree with @abenari's comment that you made the change using uuid

**Comment by isratrade** [comment]
[test]

**Comment by dmitri-d** [comment]
@isratrade: ping?

**Comment by isratrade** [comment]
@witlessbird, looks ok. I thought I previously ACK'd it.

**Comment by GregSutcliffe** [comment]
@witlessbird seems to mostly work - two small issues:

1) I use image-backed libvirt VMs most of the time, rather than PXE, and it's not cloning the image used to create the base host. In actual fact, I'm not sure we store it anywhere once the first VM was created, but the JS should at least be refreshed to select the default Image for the OS, as when provisioning a VM normally

2) VMs aren't starting on boot - for some reason the "Start (Power ON this machine)" checkbox is missing on the VM tab when cloning. Once I start the cloned machine, it boots fine. This may be related to the image-based provisioning I'm doing

I can give you access to my setup if you want to see the issue first-hand.

**Comment by GregSutcliffe** [comment]
After a joint debug session, we've fixed (2) and decided to leave (1). In (1) it's only one click to select the correct image anyway, and if the user has selected a compute profile on the primary tab, then (1) will never occur as the profile will rewrite everything anyway.

:+1: to merge

**Comment by domcleal** [comment]
Sorry, it seems this has got out of sync, could you rebase please?

**Comment by dmitri-d** [comment]
@domcleal: rebased!

**Comment by domcleal** [comment]
Merged as f5ab56a for Foreman 1.5.2, thanks @witlessbird.

**Pull request closed without merging**